src/test/java/io/jenkins/plugins/report/jtreg/main/comparator/QueryStringTest.java

```diff
@@ -0,0 +1,148 @@
+package io.jenkins.plugins.report.jtreg.main.comparator;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+
+import static org.junit.Assert.fail;
+
+public class QueryStringTest {
+    private static final File[] listOfJobs = new File[10];
+
+    @BeforeClass
+    public static void setupDummyJobs() {
+        listOfJobs[0] = new File("jtreg~full-jp17-ojdk17~rpms-f36.x86_64-fastdebug.sdk-f36.x86_64.testfarm-x11.shenandoah.ignorecp.lnxagent.jfroff");
```

Comment:
this sounds like a design flaw.
Maybe it would be better to creat some interface JobsProvider - right now it will have onloy oe implementation, listing dirs/files in some directory.
But eg for thhe purpse of testing, it will have NoFsPorivder impl, which would operate above simpel arraylist ofs trings.

it wil nto eliminate need to test over real files, but I thnk the general idea should be clear (and appicable also elsewhere).

The applicvation itself will fo course remin on FsJobLIsting, as you need to step insinde the directories. A